x/harvest/keeper/liquidation_test.go

```diff
@@ -0,0 +1,296 @@
+package keeper_test
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto"
+	tmtime "github.com/tendermint/tendermint/types/time"
+
+	"github.com/kava-labs/kava/app"
+	"github.com/kava-labs/kava/x/harvest"
+	"github.com/kava-labs/kava/x/harvest/types"
+	"github.com/kava-labs/kava/x/pricefeed"
+)
+
+// type LiquidationTestSuite struct {
+// 	suite.Suite
+// }
+
+func (suite *KeeperTestSuite) TestKeeperLiquidation() {
+	type args struct {
+		borrower             sdk.AccAddress
+		keeper               sdk.AccAddress
+		keeperRewardPercent  sdk.Dec
+		initialModuleCoins   sdk.Coins
+		initialBorrowerCoins sdk.Coins
+		initialKeeperCoins   sdk.Coins
+		depositCoins         []sdk.Coin
+		borrowCoins          sdk.Coins
+		liquidateAfter       int64
+		auctionSize          sdk.Int
+	}
+
+	type errArgs struct {
+		expectPass bool
+		contains   string
+	}
+
+	type liqTest struct {
+		name    string
+		args    args
+		errArgs errArgs
+	}
+
+	model := types.NewInterestRateModel(sdk.MustNewDecFromStr("0"), sdk.MustNewDecFromStr("0.1"), sdk.MustNewDecFromStr("0.8"), sdk.MustNewDecFromStr("0.5"))
+	reserveFactor := sdk.MustNewDecFromStr("0.05")
+
+	// oneDayInSeconds := int64(86400)
+	// oneWeekInSeconds := int64(604800)
+	oneMonthInSeconds := int64(2592000)
+	// oneYearInSeconds := int64(31536000)
+
+	borrower := sdk.AccAddress(crypto.AddressHash([]byte("testborrower")))
+	keeper := sdk.AccAddress(crypto.AddressHash([]byte("testkeeper")))
+
+	testCases := []liqTest{
+		{
+			"valid: keeper liquidates borrow",
```

Comment:
tests to add:

1) invalid: not above ltv // liquidation fails because borrow position is not above required ltv
2) valid: single deposit, multiple borrows
3) valid: multiple deposits, single borrow
4) valid: mutliple stablecoin deposits, multiple variable coin (btc, bnb, etc) borrows

Reply:
Added 1, 2 in https://github.com/Kava-Labs/kava/pull/731/commits/32418f773561c8b2fd8e788923ff06f57665c4db. Getting an error on #3 which I'm debugging. The commit includes the commented out stub of the test case that's causing the error.

Reply:
Added tests for 3, 4 in https://github.com/Kava-Labs/kava/pull/731/commits/d37a0d453bba4c2cc5faeae45b7287eeeab27214